volatility3/framework/plugins/windows/ldrmodules.py

```diff
@@ -82,45 +88,115 @@ def filter_function(x: interfaces.objects.ObjectInterface) -> bool:
+                        )
+                    if mem_mod:
+                        mem = "{0} : {1}".format(
+                            mem_mod.FullDllName.get_string(),
+                            mem_mod.BaseDllName.get_string(),
+                        )
+                except exceptions.InvalidAddressException:
+                    continue
+
+                if not self.config.get("verbose", True):
+                    yield (
+                        0,
+                        [
+                            int(proc.UniqueProcessId),
+                            str(
+                                proc.ImageFileName.cast(
+                                    "string",
+                                    max_length=proc.ImageFileName.vol.count,
+                                    errors="replace",
+                                )
+                            ),
+                            format_hints.Hex(base),
+                            load_mod is not None,
+                            init_mod is not None,
+                            mem_mod is not None,
+                            mapped_files[base],
+                        ],
+                    )
+                else:
+                    yield (
+                        0,
+                        [
+                            int(proc.UniqueProcessId),
+                            str(
+                                proc.ImageFileName.cast(
+                                    "string",
+                                    max_length=proc.ImageFileName.vol.count,
+                                    errors="replace",
+                                )
+                            ),
+                            format_hints.Hex(base),
+                            load_mod is not None,
+                            init_mod is not None,
+                            mem_mod is not None,
+                            mapped_files[base],
+                            load,
+                            init,
+                            mem,
+                        ],
+                    )
 
     def run(self):
         filter_func = pslist.PsList.create_pid_filter(self.config.get("pid", None))
         kernel = self.context.modules[self.config["kernel"]]
 
-        return renderers.TreeGrid(
-            [
-                ("Pid", int),
-                ("Process", str),
-                ("Base", format_hints.Hex),
-                ("InLoad", bool),
-                ("InInit", bool),
-                ("InMem", bool),
-                ("MappedPath", str),
-            ],
-            self._generator(
-                pslist.PsList.list_processes(
-                    context=self.context,
-                    layer_name=kernel.layer_name,
-                    symbol_table=kernel.symbol_table_name,
-                    filter_func=filter_func,
-                )
-            ),
-        )
+        if not self.config.get("verbose", True):
```

Comment:
Changing column outputs is doable, but it potentially makes it tricky for things that use the output of plugins automatically (nothing does that I know of yet, and they could read the data from the `TreeGrid` they get back, but...).  If the extra data isn't too messy, then I'd just output it all, all the time.  If it is, then we can go with the optional columns, but again, we're trying to return things as if we're in a database, so separation of things that are separate and as little duplication as possible...  5:)